cmd/cli/app/project/role/role_grant.go

```diff
@@ -40,22 +41,44 @@ func GrantCommand(ctx context.Context, cmd *cobra.Command, conn *grpc.ClientConn
 	client := minderv1.NewPermissionsServiceClient(conn)
 
 	sub := viper.GetString("sub")
+	email := viper.GetString("email")
+	pu := viper.GetString("preferred-username")
 	r := viper.GetString("role")
 	project := viper.GetString("project")
 
 	// No longer print usage on returned error, since we've parsed our inputs
 	// See https://github.com/spf13/cobra/issues/340#issuecomment-374617413
 	cmd.SilenceUsage = true
 
-	_, err := client.AssignRole(ctx, &minderv1.AssignRoleRequest{
+	req := &minderv1.AssignRoleRequest{
 		Context: &minderv1.Context{
 			Project: &project,
 		},
 		RoleAssignment: &minderv1.RoleAssignment{
-			Role:    r,
-			Subject: sub,
+			Role: r,
 		},
-	})
+	}
+
+	if sub != "" {
+		req.RoleAssignment.Subject = sub
+	} else {
+		req.RoleAssignment.Mapping = &minderv1.RoleAssignment_Mapping{
+			ClaimsToMatch: &structpb.Struct{
+				Fields: map[string]*structpb.Value{},
+			},
```

Comment:
that returns an error since it does validation on the map it takes as an argument. We'd then need to add error handling which would effectively be noop.